### ISSUE TITLE
Add hover/highlight styles on data interaction

### ISSUE THREAD

**Issue by ryanbaumann** [mapbox/mapboxgl-powerbi#28]
# Feature Request

Add hover/highlight effects on mouseover and data selection for circle, cluster, and choropleth visualization types.  Use the new GL JS `Feature States` functionality to achieve this - here's an [example](https://github.com/mapbox/mapbox-gl-js/blob/60aaa4f0e1b5412e0cffb77ded5d3dabcc1b07a9/docs/pages/example/hover-styles.html#L27) of how this works.  

Note that feature states require a top level `id` integer parameter to be specified, like this:

```json
{
    "type": "Feature",
    "id": 1,
    "geometry": {
        "type": "Point",
        "coordinates": [0, 0]
    },
    "properties": {
        "color": "blue"
    }
}
```

# To Do

- [x] Merge the latest `beta-1.2` build of GL JS into this mapboxgl-PowerBI project to gain `Feature State` functionality @ryanbaumann
- [x] (canceled) Insert a top level unique`id`value for each row from the Power BI data model into the geojson data source passed to Mapbox GL JS (see example above).  This `id` top-level key must be a unqiue value among all the features in that source.  Use the row number in the data source as a good starting point.
- [x] Add a `hover` layer for `circle` and `choropleth` viz layers that specifies a hover-state style when the user hovers their mouse cursor over a feature.  Use a layer filter to toggle the features displayed in the `hover` layer.

Use this example as a guide:
    * https://www.mapbox.com/mapbox-gl-js/example/hover-styles/

This issue compliments the data-selection & filter map issue here -> https://github.com/mapbox/mapboxgl-powerbi/issues/2

**Comment by szilardhuber** [comment]
@ryanbaumann where can we find the documentation for this new API? For choropleth we use vector source type while in the example geojson source is used? Does this `setFeatureState` method work together with vector sources as well?

**Comment by ryanbaumann** [comment]
@szilardhuber yes, the new setFeatureStatue method works on vector sources as well.  More docs are incoming soon.

Here's an [example](https://github.com/mapbox/mapbox-gl-js/blob/60aaa4f0e1b5412e0cffb77ded5d3dabcc1b07a9/docs/pages/example/data-join.html#L81) of how it works with the data-join technique.  cc/ @asheemmamoowala

**Comment by ryanbaumann** [comment]
Since feature states are still in an early Alpha phase in GL JS, let's move forward on this issue by removing the requirement to use FeatureStates.  Instead, we will use layer filters.  When a user hovers over a feature, we will add a new layer over the top which has a `hover` style state.  I'll provide an example of how this works tomorrow.

**Comment by ryanbaumann** [comment]
@endre-jenei updated the original post with the new layer filter technique list of tasks.

**Comment by samgehret** [comment]
Tested this in the drilldown branch, this looks to be working. See screenshot
![screen shot 2018-06-06 at 12 36 42 pm](https://user-images.githubusercontent.com/3859246/41061120-756a6b30-6986-11e8-8091-059ecc15e03e.png)

@ryanbaumann

**Comment by szilardhuber** [comment]
> One bug:
>
> When three drilldown layers are added to the map, I am unable to hover on the 2nd or 3rd drill level. One side effect of this is after I SHIFT + Click on a specific state to drill down to county, I am not able to SHIFT + Click on the country (which is the next drill down level).

It should be fixed with 7ba6e9a0c229d1acedf7f1f6ba59687c93558300

**Comment by ryanbaumann** [comment]
This is complete in https://github.com/mapbox/mapboxgl-powerbi/pull/163
